src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -50,28 +48,36 @@ namespace Stratis.Bitcoin.Features.Miner
     /// </remarks>
     public class PosMinting
     {
-        public class StakeOutput
-        {
-            public UtxoStakeDescription UtxoStakeDescription;
-            public int Depth;
-        }
-
+        /// <summary>
+        /// Information related to UTXO that is required for staking.
+        /// </summary>
         public class UtxoStakeDescription
         {
-            public uint256 HashBlock;
-            public TxOut TxOut;
-            public OutPoint OutPoint;
-            public int OutputIndex;
-            public HdAddress Address;
-            public UnspentOutputs UtxoSet;
-            public WalletSecret Secret;
-            public Key Key;
+            /// <summary>Block's hash.</summary>
+            public uint256 HashBlock { get; set; }
+            /// <summary>UTXO.</summary>
```

Comment:
can we be little more descriptive here?
i'm little bit confused here with this TxOut vs what is below as UtxoSet
as i described elsewhere (was in the other PR i think), i'm somewhat convinced that thing should not be called UtxoSet, 
i'm not sure if i'm sure about that :-/

what is the difference between TxOut and UnspentOutputs?
TxOut seems to to refer to a single UTXO inside UnspentOutputs, which represents selected outputs of a transactions

the question is whether we can somehow provide a comment here that would make it easier for the reader to understand that logic here

although i know this is difficult, those structures are not trivial to understand, what they actually represent